### PR TITLE
[DROOLS-4513] - s2i build fails at KieServerContainerVerifier

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: Filippe Spolti <fspolti@redhat.com>

### CODE REVIEW

File: drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/InternalKieModule.java [hunk 1]
```diff
@@ -146,13 +148,14 @@ default ProjectClassLoader createModuleClassLoader( ClassLoader parent ) {
...
+    static InternalKieModule createKieModule(ReleaseId releaseId, File jar) {
+        try (ZipFile zipFile = new ZipFile(jar)) {
             ZipEntry zipEntry = zipFile.getEntry(KieModuleModelImpl.KMODULE_JAR_PATH);
             KieModuleModel kieModuleModel = KieModuleModelImpl.fromXML(zipFile.getInputStream(zipEntry));
             setDefaultsforEmptyKieModule(kieModuleModel);
-            return kieModuleModel != null ? InternalKieModuleProvider.get(adapt( releaseId ), kieModuleModel, jar) : null;
-        } catch ( Exception e ) {
+            return kieModuleModel != null ? InternalKieModuleProvider.get(adapt(releaseId), kieModuleModel, jar) : null;
+        } catch (Exception e) {
+            LocalLogger.logger.error(e.getMessage(), e);
```
Comment (mswiderski) [inline drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/InternalKieModule.java:158]: so in the end just logging and not throwing? what was the reason for it?

**Comment by mswiderski** [comment]
@spolti would be good to link this PR with the one in droolsjbpm-knowledge repo since they refer to the same jira

File: drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/InternalKieModule.java (hunk 1 above)
Comment (baldimir) [reply drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/InternalKieModule.java:158]: I am also interested in this.

**Comment by spolti** [comment]
Related PR:  https://github.com/kiegroup/droolsjbpm-knowledge/pull/402

File: drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/InternalKieModule.java (hunk 1 above)
Comment (spolti) [reply drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/InternalKieModule.java:158]: @mswiderski I sent you email to you and @mariofusco, I'll add it's content here for a broader discussion.

Mario had a concern about add a thrown new Exception here:
https://github.com/kiegroup/drools/blob/master/drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/InternalKieModule.java#L155

By adding a thrown there I got the messages I need on KieContainerVerifier, but, some tests starts to fail, like:
https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/drools-pullrequests/348/testReport/junit/org.drools.example.api.kiecontainerfromkierepo/KieContainerFromKieRepoExampleTest/testGo/

If I add a try/catch block around this line
https://github.com/kiegroup/drools/blob/master/drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/InternalKieModule.java#L152
I am not sure why, but the catch block is not called.


Another possible solution I found was:

    static InternalKieModule createKieModule(ReleaseId releaseId, File jar) {
        try (ZipFile zipFile = new ZipFile(jar)) {
            ZipEntry zipEntry = zipFile.getEntry(KieModuleModelImpl.KMODULE_JAR_PATH);
            KieModuleModel kieModuleModel = KieModuleModelImpl.fromXML(zipFile.getInputStream(zipEntry));
            setDefaultsforEmptyKieModule(kieModuleModel);
            return kieModuleModel != null ? InternalKieModuleProvider.get(adapt(releaseId), kieModuleModel, jar) : null;
        } catch (Exception e) {
            if (e instanceof NullPointerException) { // or another catch block
                return null;
            }
            throw new RuntimeException(e.getMessage(), e);
        }
    }


This way I still getting the error messages, the KieContainerFromKieRepoExampleTest/testGo/ stops to complain but another test failed:

https://github.com/kiegroup/drools/blob/master/kie-ci/src/test/java/org/kie/scanner/KieRepositoryScannerTest.java#L237

with the stacktrace: https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/drools-pullrequests/350/testReport/junit/org.kie.scanner/KieRepositoryScannerTest/testKieScannerScopesNotRequired_Manually_wired_component_provider_false_/


With that said, it confirm the Marios' concern about use a thrown on tha try/catch.

So, for now I added a logger there, not sure if there is a better way to log stuff on interfaces:

```
} catch (Exception e) {
    LocalLogger.logger.error(e.getMessage(), e);
}

....

final class LocalLogger {
    private static final Logger logger = LoggerFactory.getLogger(InternalKieModule.class);
}

```


Besides that, we could refactor the createKieModule method to not use try with resources and deal with specific cases on different catch blocks.

What do you guys think?

**Comment by spolti** [comment]
@mswiderski @mariofusco did you have time to take a look on my comment? thoughts?

File: drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/InternalKieModule.java (hunk 1 above)
Comment (mswiderski) [reply drools-compiler/src/main/java/org/drools/compiler/kie/builder/impl/InternalKieModule.java:158]: thanks for detailed explanation @spolti 

if that solves the original issue then I think we are good to go here.

**Review by mswiderski — APPROVED** [review]

**Review by mariofusco — APPROVED** [review]

**Pull request merged**